spec/lib/discourse_spec.rb

```diff
@@ -340,6 +340,32 @@ def call(ex, ctx)
       Sidekiq.error_handlers.delete(logger)
     end
 
+    it "should collect job exception stats" do
+      Discourse.reset_job_exception_stats!
+
+      # see MiniScheduler Manager which reports it like this
+      exception_context = {
+        message: "Running a scheduled job",
+        job: { "class" => Jobs::ReindexSearch }
+      }
+
+      # re-raised unconditionally in test env
+      Discourse.handle_job_exception(StandardError.new, exception_context) rescue nil
+      Discourse.handle_job_exception(StandardError.new, exception_context) rescue nil
```

Comment:
```suggestion
      2.times do
        expect { Discourse.handle_job_exception(StandardError.new, exception_context) }.to raise_error(StandardError)
      end
```

I think we should generally avoid `rescue nil`. Although the following assertion will fail if there is a typo, it doesn't immediately point to where the problem is.